contracts/TracerPerpetualSwaps.sol

```diff
@@ -204,24 +209,44 @@ contract TracerPerpetualSwaps is
         Balances.Account storage account1 = balances[order1.maker];
         Balances.Account storage account2 = balances[order2.maker];
 
+        int256 fee =
+            PRBMathUD60x18
+                .mul(uint256(quoteChange), uint256(feeRate.toInt256()))
+                .toInt256();
+
         /* TODO: handle every enum arm! */
         if (order1.side == Perpetuals.Side.Long) {
             // user 1 is long. Increase base, decrease quote
-            account1.position.quote = account1.position.quote - quoteChange;
```

Comment:
Just flagging this for review and future consideration, but this function could largely be extracted out into a pure library function that operates solely on `Balances.Position`.